docs/Protected_build_porting_guide.md

```diff
@@ -0,0 +1,363 @@
+	- [Creating new configuration files](#Creating-new-configuration-files)
+	- [Creating linker scripts](#Creating-linker-scripts)
+	- [Userspace initialization](#Userspace-initialization)
+	- [Mpu initialization](#Mpu-initialization)
+	- [Allocation of kernel and user heaps](#Allocation-of-kernel-and-user-heaps)
+	- [Changes in make.defs and makefile](#Changes-in-make.defs-and-makefile)
+	- [Changes in the download method](#Changes-in-the-download-method)
+
+
+## Introduction
+This is a guide for porting Protected Build for a new chipset.
+
+### Assumptions
+- TizenRT 3.0 flat build has been successfully ported and tested on the target CHIPSET board
+- Generic code relating to TizenRT protected build is already present in the codebase (MACRO: CONFIG_ BUILD_PROTECTED)
+- In order to bring up protected build, changes are to be made specific to the board
+- Information is available about the memory map of the board, partition information and memory organization
+
+### What is Protected Build
+- In protected build, two separate binaries namely kernel binary and user binary are created.
+- The kernel binary is created under the name - tinyara.bin and user binary is placed as - tinyara_user.bin in the <build/output/bin> folder
+- The kernel binary consists of the following modules - architecture(ARCH), core Kernel Module(SCHED), C-Lib and memory manager (MM), file system (FS) and the drivers.
+- The user binary includes - Applications, C-Lib and Memory Manager (MM), Framework and External modules.
+- C-lib and Memory Manager are duplicated in the kernel space and the user space.
+- The two binaries are placed in separate memory regions and run using separate RAM memories.
+- The code and data memories on the board are split into two parts for use by user and kernel.
+- For example - the FLASH memory of the board is split into kernel-flash and user-flash and same is the case for RAM memory.
+- If the architecture supports MPU (Memory protection Unit), kernel memory region is protected and the processor has access only to user-space memory areas while running  under unprevileged mode.
+- User-space can access the kernel services through System Calls (downward communication)
+- Some information about user-space such as memory regions, and some user-space API's needs to be shared to kernel.
+- This information is stored in a special data structure called as the User Space Object which is linked into the kernel during build time.
+- The kernel can use the function pointers to user-space APIs available in the user space object to invoke required user space funtionality.
+
+
+## Steps for Porting Protected Build in a new chipset
+### Create a new configuration for Protected Build
+- Make a copy of the existing flat build configuration and rename it for protected build
+- For example: every chipset has a hello configuration under path build/config/CHIPSET_NAME.
```

Comment:
As said, dont plan this to be a standalone porting guide. This should be plugged into TizenRT porting guide.

Reply:
This document is plugged into TizenRT porting guide as mentioned.